crawler/mesos.py

```diff
@@ -0,0 +1,237 @@
+    'master/slaves_disconnected': Stat("gauge", "master/slaves_disconnected"),
+    'master/slave_shutdowns_canceled': Stat("counter", "master/slave_shutdowns_canceled"),
+    'master/slave_shutdowns_scheduled': Stat("counter", "master/slave_shutdowns_scheduled"),
+    'master/task_lost/source_master/reason_slave_removed': Stat("counter", "task_lost/source_master/reason_slave_removed"),
+    'master/tasks_error': Stat("counter", "master/tasks_error")
+}
+
+# FUNCTION: gets the list of stats based on the version of mesos
+def get_stats_string(version):
+    if version == "0.19.0" or version == "0.19.1":
+       stats_cur = dict(STATS_MESOS.items() + STATS_MESOS_019.items())
+    elif version == "0.20.0" or version == "0.20.1":
+       stats_cur = dict(STATS_MESOS.items() + STATS_MESOS_020.items())
+    elif version == "0.21.0" or version == "0.21.1":
+       stats_cur = dict(STATS_MESOS.items() + STATS_MESOS_021.items())
+    elif version == "0.22.0" or version == "0.22.1":
+       stats_cur = dict(STATS_MESOS.items() + STATS_MESOS_022.items())
+    else:
+       stats_cur = dict(STATS_MESOS.items() + STATS_MESOS_022.items())
+
+    return stats_cur
+
+
+def configure_crawler_mesos(inurl):
+    logger.debug('Mesos url %s' % inurl)
+    CONFIGS.append({
+        'mesos_url': inurl[0]
+    })
+
+def fetch_stats(mesos_version):
+    if CONFIGS == []:
+       CONFIGS.append({
+          'mesos_url': 'http://localhost:5050/metrics/snapshot'
+        })
+    logger.debug('connecting to %s' % CONFIGS[0]['mesos_url'])
+    print CONFIGS[0]['mesos_url']
+    try:
+        result = json.loads(urllib2.urlopen(CONFIGS[0]['mesos_url'], timeout=10).read())
+    except urllib2.URLError, e:
+        logger.exception('Exception opening mesos url %s',None)
+        return None
+    logger.debug('mesos_stats %s' % result)
+    print "result"
+    print result
+    return result
+
+
+def setup_logger(logger_name, logfile='crawler.log', process_id=None):
+    _logger = logging.getLogger(logger_name)
+    _logger.setLevel(logging.DEBUG)
+    (logfile_name, logfile_xtnsion) = os.path.splitext(logfile)
+    if process_id is None:
+        fname = logfile
+    else:
+        fname = '{0}-{1}{2}'.format(logfile_name, process_id,
+                                    logfile_xtnsion)
+    h = logging.handlers.RotatingFileHandler(filename=fname,
+                                             maxBytes=10e6, backupCount=1)
+    f = logging.Formatter(
+        '%(asctime)s %(processName)-10s %(levelname)-8s %(message)s')
+    h.setFormatter(f)
+    _logger.addHandler(h)
+
+
+def dig_it_up(obj, path):
+    try:
+        if type(path) in (str, unicode):
+            path = path.split('.')
+        return reduce(lambda x, y: x[y], path, obj)
+    except:
+        return False
+
+
+def log_verbose(enabled, msg):
+    if not enabled:
+        return
+    logger.debug('mesos-master plugin [verbose]: %s' % msg)
+
+
+def snapshot_crawler_mesos_frame(inurl):
```

Comment:
If you call this function from your test also,  you can increase coverage considerably.